[lloyal-ai-inlined-vector] new port

zuhair-naqvi opened the pull request:
### Description
Adds inlined-vector v5.7.1 - A modern C++17/20 header-only SBO vector designed for mobile/embedded deployment where bundling Abseil or Boost isn't viable.

**Why this port is useful:**
- **Unique capability**: Only SBO implementation that compiles `insert()`/`erase()` for types with `const` members (non-assignable types). `std::vector`, `absl::InlinedVector`, and `boost::small_vector` all fail to compile these cases.
- **Zero dependencies**: Single 800-line implementation (1030 total with comments) - no ecosystem required
- **Competitive performance**: 13.2× faster than `std::vector` for inline operations; within 3% on heap paths despite rebuild-and-swap strategy

**Differentiation from alternatives:**
- vs `std::vector`: Adds SBO (zero heap allocations for size ≤ N)
- vs `absl::InlinedVector`: Zero dependencies; supports non-assignable types; bidirectional heap↔inline transitions
- vs `boost::small_vector`: Single header; doesn't permanently lock to heap after capacity changes; supports non-assignable types
- vs `llvm::SmallVector`: Modern C++17/20 (uses `std::variant`); zero dependencies; non-assignable type support.

**Architecture:**
- Uses `std::variant<InlineStorage, HeapStorage>` for cleaner state management vs hand-rolled unions
- Rebuild-and-swap for heap `insert()`/`erase()` enables non-assignable type support with negligible overhead
- Parent pointer architecture for zero-overhead allocator propagation (even for inline elements)

**Performance (Apple M1, N=16, -O3, vs Abseil master / Boost 1.88):**
- Inline trivial fills (N=8): **13.2× faster** than `std::vector`
- Inline string fills (N=8): 1.4× faster than `std::vector` (within 10% of Abseil)
- Heap inserts (N=128): **Fastest** (6169ns vs 6335ns Abseil) - rebuild-and-swap wins at scale
- Move construction: Fastest among all tested implementations

  ### New Port Checklist

  - [x] Changes comply with the [maintainer 
  guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
  - [x] The name of the port matches an existing name for this component on https://repology.org/ if possible,
  and/or is strongly associated with that component on search engines.
  - [x] Optional dependencies are resolved in exactly one way. (N/A - no dependencies)
  - [x] The versioning scheme in `vcpkg.json` matches what upstream says.
  - [x] The license declaration in `vcpkg.json` matches what upstream says.
  - [x] The installed as the "copyright" file matches what upstream says.
  - [x] The source code of the component installed comes from an authoritative source.
  - [x] The generated "usage text" is accurate. See
  [adding-usage](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/examples/adding-usage.md) for context.
  - [ ] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
  - [x] Only one version is in the new port's versions file.
  - [ ] Only one version is added to each modified port's versions file.

  ### Port Details
  - **Type**: Header-only library
  - **License**: MIT
  - **Homepage**: https://github.com/lloyal-ai/inlined-vector
  - **Dependencies**: None (C++17 STL only)
  - **Version**: 5.7.1

  ### Key Features
  - Zero external dependencies (single 1030-line header)
  - 13.2× faster than std::vector for inline operations (trivial types, N=8)
  - Full allocator support (std::pmr compatible, POCMA/POCS/SOCCC compliant)
  - Supports non-assignable types (unique capability vs std::vector/abseil/boost)
  - Bidirectional heap↔inline transitions via shrink_to_fit()
  - Production-ready: 15/15 unit tests + 9/9 fuzz tests passing, zero ASan/UBSan violations

  ### Testing
  - [x] Built and tested locally on macOS (Apple M1, AppleClang 17.0)
  - [x] All CI checks passing on source repository
  - [x] License file included
  - [x] Usage file provided
  - [x] Port follows vcpkg guidelines

  ### Note on Version Database
  The version database update (`./vcpkg x-add-version --all`) will be run after initial review feedback, as is
  standard practice for new ports to avoid conflicts with concurrent PRs.

  ### Verification
  Release tarball SHA512:
  619777f8dd930813e5be96cdfa5171485356fb8ac6ed3f32fca3ad68565a48c269a38a022d734b118c0e397f5f84ad591a27c42b7080616af5d7d0575e6a7a9b

zuhair-naqvi committed:
[inlined-vector] new port

zuhair-naqvi commented:
@microsoft-github-policy-service agree company="lloyal.ai"

zuhair-naqvi committed:
Update version database and format manifest

zuhair-naqvi committed:
Add vcpkg-cmake dependencies for CMake helper functions

BillyONeal commented on ports/inlined-vector/vcpkg.json:
```diff
@@ -0,0 +1,19 @@
+{
+  "$schema": "https://raw.githubusercontent.com/microsoft/vcpkg-tool/main/docs/vcpkg.schema.json",
```
```suggestion
```

BillyONeal reviewed (changes requested):
The only 'seriously blocking' comment here is the following one about naming but would like to see the other nitpicks at least considered. Thanks for the new port! Please mark "Ready for review" when you're ready for us to look again.

> * [x]  The name of the port matches an existing name for this component on https://repology.org/ if possible,
>   and/or is strongly associated with that component on search engines.

As far as I can tell this was checked incorrectly. The name `inlined-vector` is not in repology, and if I search for it I actually get abseil's implementation first, not this one.

Would you accept changing the name of the port to `lloyal-ai-inlined-vector` ? Note that I am only speaking about the vcpkg port, the name of the library, what it installs, etc. is fine.

zuhair-naqvi committed:
[lloyal-ai-inlined-vector] Address reviewer feedback

- Rename port from inlined-vector to lloyal-ai-inlined-vector
- Add VCPKG_BUILD_TYPE release declaration for header-only port
- Remove explicit lib directory cleanup (unnecessary with build type set)
- Simplify usage file (remove sample code per vcpkg policy)
- Update version database with new port name

Addresses feedback from @BillyONeal in PR #47960

zuhair-naqvi committed:
Fix: Remove lib directory after config_fixup

The CMakeLists.txt installs CMake config files to lib/cmake/inlined-vector.
After vcpkg_cmake_config_fixup moves them, the empty lib/ directory must be
removed to satisfy vcpkg's ALLOW_EMPTY_FOLDERS policy.

zuhair-naqvi committed:
Update lloyal-ai-inlined-vector to v5.7.1

- Update to v5.7.1 (fixes header-only install path)
- Change CONFIG_PATH to share/inlined-vector/cmake
- Remove lib/ directory cleanup (no longer needed)
- Update SHA512 for new release tarball
- Upstream now installs to share/ correctly per header-only convention

zuhair-naqvi committed:
Fix: Use v prefix in REF for GitHub archive URL

vcpkg was trying to fetch from archive/5.7.1.tar.gz but GitHub
tags use v prefix, so the URL should be archive/v5.7.1.tar.gz

Changed REF from "${VERSION}" to "v${VERSION}"

zuhair-naqvi committed:
Remove $schema line per vcpkg conventions

vcpkg ports don't use the $schema line in vcpkg.json.
Removed to match standard vcpkg port format.

Addresses BillyONeal's suggestion.

BillyONeal committed:
Fix version DB.

BillyONeal committed:
Remove "add 1 to x" comments.

zuhair-naqvi committed:
Merge pull request #1 from BillyONeal/lloyal-ai-inlined-vector

zuhair-naqvi commented:
> The PR was not opened with permission for me to fix these nitpicks so I submitted a PR: https://github.com/lloyal-ai/vcpkg/pull/1
> 
> 
> 
> Thanks for the new port!

Merged.

Thanks for your assistance!

BillyONeal commented:
Thanks!
